### PR TITLE
Passive: hide full{value,infos} from user, move it to MongoDB specific code

### PR DESCRIPTION
**Pull request by Nutigoodpipes**
Move 'fullvalue' and 'fullinfos' functions in mongodb specific code, since they are used only for size restrictions on mongodb indexes.

### CODE REVIEW

**Comment by p-l-** [comment]
@Nutigoodpipes do you have two commits with the same title on purpose?

File: ivre/db/mongo.py [hunk 1]
```diff
@@ -3351,6 +3352,11 @@ def get(self, spec, hint=None, **kargs):
                                                 rec.pop('addr_1')])
             except (KeyError, socket.error):
                 pass
+            for key in ['value', 'targetval']:
+                if 'full' + key in rec:
+                    rec[key] = rec.pop('full' + key)
+            if 'fullinfos' in rec:
+                rec.setdefault('infos', {}).update(rec['fullinfos'])
```
Comment (p-l-) [inline ivre/db/mongo.py]: Maybe we even want to `.pop('fullinfos')`?

File: ivre/db/mongo.py [hunk 2]
```diff
@@ -3368,6 +3374,11 @@ def get_one(self, spec, **kargs):
             pass
         else:
             del rec['addr_0'], rec['addr_1']
+        for key in ['value', 'targetval']:
+            if 'full' + key in rec:
+                rec[key] = rec.pop('full' + key)
+        if 'fullinfos' in rec:
+            rec.setdefault('infos', {}).update(rec['fullinfos'])
```
Comment (p-l-) [inline ivre/db/mongo.py]: Same here.

File: ivre/db/mongo.py [hunk 3]
```diff
@@ -3376,6 +3387,32 @@ def update(self, spec, **kargs):
...
+                spec[key] = hashlib.sha1(spec[key].encode()).hexdigest()
+
+    @staticmethod
+    def _fix_infos_size(spec):
+        # We enforce a utils.MAXVALLEN // 10 size limits for subkey values in
+        # infos; this is because MongoDB cannot index values longer than 1024
+        # bytes.
+        for field in list(spec.get("infos", {})):
+            value = spec["infos"]
+            if len(value) > utils.MAXVALLEN:
```
Comment (p-l-) [inline ivre/db/mongo.py]: That does not seem consistent with the above comment (`utils.MAXVALLEN // 10`?).

Comment (p-l-) [reply ivre/db/mongo.py]: BTW, is there a reason to have two distinct methods here?

**Comment by p-l-** [comment]
BTW this would fix #618.

File: ivre/db/mongo.py (hunk 3 above)
Comment (Nutigoodpipes) [reply ivre/db/mongo.py]: Same reason I had two commits, uncleaned dev process...

Comment (Nutigoodpipes) [reply ivre/db/mongo.py]: Wrong fix now fixed.

**Pull request merged**